src/disks.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+import dataclasses
+from glob import glob
+from typing import Optional
+
+import pyudev
+import psutil
+
+from moulinette.utils.log import getActionLogger
+
+
+from yunohost.utils.disks import filter_device
+
+
+logger = getActionLogger("yunohost.storage")
+
+
+@dataclasses.dataclass
+class DiskParts:
+    devname: str
+    filesystem: str
+    encrypted: bool
+    mountpoint: str
+
+    @staticmethod
+    def from_parent_device(device: pyudev.Device, partitions):
+        result = OrderedDict()
+        for child_dev in sorted(
+            filter(filter_device, device.children), key=lambda it: it.device_node
+        ):
+            encrypted_provider = glob(f"/sys/block/dm-*/slaves/{child_dev.sys_name}")
+            if encrypted_provider:
+                # retrive the dm-x part
+                dm = encrypted_provider[0].split("/")[3]
+                enc_dev = pyudev.Devices.from_name(device.context, "block", dm)
+                # This work for LUKS, what about other partition mecanisms?
+                partname = f"/dev/mapper/{enc_dev.properties['DM_NAME']}"
+                encrypted = True
+            else:
```

Comment:
This method detects LUKS partitions but I'm not sure it does not detect other unecrypted partitions, like LVM. If someone with more system knowlege than me could check this.

Reply:
I could use [`pyparted`](https://github.com/dcantrell/pyparted) here but this needs sufficient rights. If I'm not mistaken, `yunohost-api` is running as root, so this shouldn't be an issue, right?

Reply:
Turns out this solution detects LVM (non-LUKS) partitions as encrypted. And I can't figure out how to retrieve FS with pyparted. I don't know how gparted and KDE partition manager detect the partition filesystem.